src/vsr/client_replies.zig

```diff
@@ -1,3 +1,15 @@
+//! Store the latest reply to every active client session.
+//!
+//! This allows them to be resent to the corresponding client if the client missed the original
+//! reply message (e.g. dropped packet).
+//!
+//! - Client replies' headers are redundantly stored in the SuperBlock's ClientSessions trailer.
+//! - Client replies are only stored when `header.size ≠ sizeOf(Header)`
+//!   (the header is in ClientSessions).
+//! - Corrupt client replies can be repaired from other replicas.
+//!
+//! ClientReplies may defer writing the reply immediately – often the reply will be replaced
+//! quickly. But all writes must flush during a checkpoint.
```

Comment:
I think we can maybe tell a little bit more about how things fit together, and how this relates to `ClientSessions`, something along the lines of:

Superblock's `ClientSessions` stores the _header_ of the latest reply. The body of the reply (if it is non-empty) can be fond in the `.replies` zone of the datafile through `ClientReplies` type. Replies are written to disc asynchronously and subsequent writes for the same client are coalesced (as we care only about the latest reply). We guarantee that replies are durable(*)

---

I am actually curious what's our exact durability guarantee for replies? I _think_ a cluster can lose a reply (eg, checkpointed reply gets corrupted on all replicas), but, unlike lost prepare, this won't render the cluster unavailable, right?

Reply:
> I am actually curious what's our exact durability guarantee for replies? I think a cluster can lose a reply (eg, checkpointed reply gets corrupted on all replicas), but, unlike lost prepare, this won't render the cluster unavailable, right?

Exactly right. I'll add this in a follow-up:

```zig
//! If the same reply is corrupted by all replicas, the cluster is still available.
//! If the respective client also never received the reply (due to a network fault), the client may
//! be "locked out" of the cluster – continually retrying a request which has been executed, but
//! whose reply has been permanently lost. This can be resolved by the operator restarting the
//! client to create a new session.
```